libtest/signal.cc

```diff
@@ -210,7 +214,8 @@ SignalThread::SignalThread() :
 
   sigaddset(&set, SIGUSR2);
 
-  sem_init(&lock, 0, 0);
+  strcpy(lock_name, "/XXXXXXXX");
+  mktemp(lock_name);
```

Comment:
on Ubuntu 16.4 `make` throws a lot of warnings: ```the use of `mktemp' is dangerous, better use `mkstemp' or `mkdtemp'```

Reply:
Right, this is a legit concern if you're using it to make real files. But in this case, we're making pretend filenames for semaphores. If we can suppress that warning, maybe with a pragma, we should.

Reply:
I don't think there is a way to suppress it. The warning is implemented in the linker.

Just as a note, mktemp was removed from the POSIX standard, so that could be a problem going forward.

Reply:
I know libtest isn't exactly a critical part of gearmand, but I think it would be better to revert to using sem_init on non-macOS platforms. Basically, add some `#ifdef __APPLE__` lines to libtest/signal.cc around where the semaphores are created/destroyed and bring back the old sem_init implementation in an `#else` block. Either that or add stuff to configure to test whether or not the build platform supports sem_init. What do you think?

Reply:
it sounds good to me

Reply:
I'm not against it, but I don't see the point. This works and it's just for the test suite.

Reply:
Sure, it's just the test suite and the usage of mktemp is valid, but the warning is phrased so stridently and strictly that it will give a bad impression to anyone compiling the code.